### ISSUE TITLE
How to use ATrim with Concat?

### ISSUE THREAD

**Issue by maranqz** [hudl/HudlFfmpeg#44]
I have to cut the music and made equal to the length of the video.
I need to join video files and also attach sound to them.

Incorrect filter works concat!

Thank you in advance.
```c#
 var commandFactory = CommandFactory.Create();
            var command = commandFactory.CreateOutputCommand();
            var concat = Filterchain.FilterTo<VideoStream>(new Concat(1,1));
            var sar = Filterchain.FilterTo<VideoStream>(new SetSar(new Ratio(1, 1)));

            var sm = Resource.From(inputPath + "2.mp4").LoadMetadata().Streams.OfType<VideoStream>().First().Info;
            var pad = new Pad(new System.Drawing.Size(1920, 1080), new Point(656, 1));
            pad.Color = "black,unsharp";
            var ff = Filterchain.FilterTo<VideoStream>(new Scale(1920, 1080));
            var sf = Filterchain.FilterTo<VideoStream>(new Scale(609, 1080), pad);

var first = command
        .WithInput<VideoStream>(inputPath + "1.mp4")
        .Filter(ff)
        .Filter(sar);       

    var a0 = command
        .WithInput<AudioStream>(inputPath + "1.mp3")
        .Filter(Filterchain.FilterTo<AudioStream>(
            new ATrim(null, 10, AudioUnitType.Seconds)
        )
        );
            
var second = command
    .WithInput<VideoStream>(inputPath + "2.mp4")
    .Filter(sf)
    .Filter(sar) ;

var a1 = command
        .WithInput<AudioStream>(inputPath + "2.mp3")
        .Filter(Filterchain.FilterTo<AudioStream>(
            new ATrim(null, 20, AudioUnitType.Seconds)
        )
        );
command
    .Select(first.StreamIdentifiers)
    .Select(a0.StreamIdentifiers)
    .Select(second.StreamIdentifiers)
    .Select(a1.StreamIdentifiers)
    .Filter(concat)            
    .MapTo<Mp4>(inputPath +
    inte +
    "1080.mp4", outputSettings);

command.Render();
```

Incorrect filter works.
```
 -i "path/1.mp4" -i "path/1.mp3" -i "path/2.mp4" -i "path/2.mp3" -filter_complex " [0:v]scale=w=1920:h=1080 [mpd1c8f485]; [mpd1c8f485] setsar=sar=1/1 [mp2bdbd193]; [1:a] atrim=end=10 [mpf8d320cb]; [2:v] scale=w=609:h=1080,pad=w=1920:h=1080:x=656:y=1:color=black,unsharp [mpb59b3366]; [mpb59b3366] setsar=sar=1/1 [mp69075c65]; [3:a] atrim=end=20 [mpd2b74d0a]; [mp2bdbd193] [mpf8d320cb] [mp69075c65] [mpd2b74d0a] concat=n=4:a=1 [mpeb0035ff]" -s 1920x1080 -map [mpeb0035ff] "path/6982857521080.mp4"
```

Correct filter works.
```
 -i "path/1.mp4" -i "path/1.mp3" -i "path/2.mp4" -i "path/2.mp3" -filter_complex " [0:v]scale=w=1920:h=1080 [mpd1c8f485]; [mpd1c8f485] setsar=sar=1/1 [mp2bdbd193]; [1:a] atrim=end=10 [mpf8d320cb]; [2:v] scale=w=609:h=1080,pad=w=1920:h=1080:x=656:y=1:color=black,unsharp [mpb59b3366]; [mpb59b3366] setsar=sar=1/1 [mp69075c65]; [3:a] atrim=end=20 [mpd2b74d0a]; [mp2bdbd193] [mpf8d320cb] [mp69075c65] [mpd2b74d0a] concat=n=2:a=1 [mpeb0035ff] [audio]" -s 1920x1080 -map [mpeb0035ff] -map [audio] "path/6982857521080.mp4"
```

**Comment by Casey-Bateman** [comment]
@maranqz So what you could do is run two concat filters, one for audio and one for video. then map those both together. I have used this approach before.

**Comment by maranqz** [comment]
Lead by example please, and I do not understand.

**Comment by Casey-Bateman** [comment]
Adding this example to samples application available in version `v3.4.0`

**Comment by Casey-Bateman** [comment]
Added under samples directory
